File: support_tools/swigwin-3.0.12/Examples/test-suite/go/virtual_poly_runme.go
package main

import "./virtual_poly"

func main() {
	d := virtual_poly.NewNDouble(3.5)
	i := virtual_poly.NewNInt(2)

	// the copy methods return the right polymorphic types
	dc := d.Copy()
	ic := i.Copy()

	if d.Get() != dc.Get() {
		panic(0)
	}

	if i.Get() != ic.Get() {
		panic(0)
	}

	virtual_poly.Incr(ic)

	if (i.Get() + 1) != ic.Get() {
		panic(0)
	}

	dr := d.Ref_this()
	if d.Get() != dr.Get() {
		panic(0)
	}

	// 'narrowing' also works
	ddc := virtual_poly.NDoubleNarrow(d.Nnumber())
	if d.Get() != ddc.Get() {
		panic(0)
	}

	dic := virtual_poly.NIntNarrow(i.Nnumber())
	if i.Get() != dic.Get() {
		panic(0)
	}
}

File: support_tools/swigwin-3.0.12/Examples/test-suite/python/virtual_poly_runme.py
import virtual_poly

d = virtual_poly.NDouble(3.5)
i = virtual_poly.NInt(2)

#
# the copy methods return the right polymorphic types
#
dc = d.copy()
ic = i.copy()

if d.get() != dc.get():
    raise RuntimeError

if i.get() != ic.get():
    raise RuntimeError

virtual_poly.incr(ic)

if (i.get() + 1) != ic.get():
    raise RuntimeError


dr = d.ref_this()
if d.get() != dr.get():
    raise RuntimeError


#
# 'narrowing' also works
#
ddc = virtual_poly.NDouble_narrow(d.nnumber())
if d.get() != ddc.get():
    raise RuntimeError

dic = virtual_poly.NInt_narrow(i.nnumber())
if i.get() != dic.get():
    raise RuntimeError
